src/aggregation/Aggregator.ts

```diff
@@ -22,6 +22,7 @@ import * as AggregatorFactory from './AggregatorFactory';
 /**
  * Base interface for all aggregators.
  */
+// eslint-disable-next-line
```

Comment:
After I updated the eslint plugin, it gave linting errors about unused variables. I could not understand why we have a <R> generic type here which is not used in this interface. So I had to add a eslint disable line at the top. Feel free suggest another solution

Reply:
It is probably ported from the Java side, in which we use the `R`, but it is required for the server-side. https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/aggregation/Aggregator.java#L84

We probably don't need this in the client side. I am not sure whether or not removing the generic type parameter can be counted as backward-compatible change, so lets leave it like that

Reply:
Is this backward compatibility related to client or server? Plus, by backward compatible which versions do you mean? last major version, that is 4.x, maybe? So any 4.x client should be compatible with features in older versions of 4.x?

Reply:
I mean the client. 

What we cannot be sure of is that some users might be using this public interface (or the MapEventListener) (I highly doubt there are any), with the type parameters. If so, their code won't compile when they upgrade the client version

Reply:
Reverted interface changes.